src/app/group/group-add-edit/group-add-edit.component.ts

```diff
@@ -11,68 +14,164 @@ import { filter } from 'rxjs';
+  constructor(private router: Router,
+    private activatedRoute: ActivatedRoute,
+    private groupsService: GroupsService) {
+
+    /*******************************testing code*************************************/
+
+    // this.groupsService.logoutUser().subscribe((res)=> console.log(res))
+    // setTimeout(() => {
+    //   this.getLogin();
+    // }, 1000);
+
+    // setTimeout(() => {
+    //   this.getCurrentUser()
+    // }, 2000);
+
+  };
+
+
+
+
+
 
   // ngOnInit method
   ngOnInit(): void {
+    /** Form group for add group page
+     */
     this.addGroupForm = new FormGroup({
       groupName: new FormControl('', [Validators.required]),
-      shareLink: new FormControl(''),
-      groupType: new FormControl('Home'),
+      about: new FormControl(''),
       groupMember: new FormArray([
-        this.memberNameAndEmailForm()
+        new FormGroup({
+          userId: new FormControl('3a0ba79a-0f40-0dd1-4e91-9ed501777180')
+        })
       ]),
     })
 
+
+    // form group add members to group
+    this.addMembersForm = new FormGroup({
+    groupMembers: new FormArray([
+      this.addMembersToGroupForm()
+    ]),})
+
+
     // get activatedRoute parameter using observable
-    this.activatedRoute.params.subscribe((param) =>{
+    this.activatedRoute.params.subscribe((param) => {
       this.getActivatedRouteParam = param['routerParam'];
-      console.log(this.getActivatedRouteParam)
-      if(this.getActivatedRouteParam === undefined ){
+      // console.log(this.getActivatedRouteParam)
+      if (this.getActivatedRouteParam === undefined) {
         this.isGroupAddActive = true;
       }
-      else{
+      else {
         this.isGroupAddActive = false;
       }
     })
+
+
   }
 
   // Add member name and member email formgroup in add group form
-  memberNameAndEmailForm(): FormGroup{
+  addMembersToGroupForm(): FormGroup {
     return new FormGroup({
-      memberName : new FormControl(''),
-      memberEmail : new FormControl('')
+      memberName: new FormControl(''),
+      memberEmail: new FormControl('')
     })
   }
 
   // Getter methods
-  get groupName(){
-    return this.addGroupForm.get('groupName');
+  get groupName() {
+    return this.addGroupForm.get('groupName') as FormControl;
+  }
+
+  get about() {
+    return this.addGroupForm.get('about') as FormControl;
   }
 
   // getter of dyanamic create group member
-  get memberNameEmail(){
-    return this.addGroupForm.get('groupMember') as FormArray;
+  get groupMembers() {
+    return this.addMembersForm.get('groupMembers') as FormArray;
   }
 
   // Add member dynamically
-  addMemberNameAndEmail(): void{
-    this.memberNameEmail.push(this.memberNameAndEmailForm());
+  addMemberNameAndEmailField(): void {
+    this.groupMembers.push(this.addMembersToGroupForm());
   }
 
   // Delete member name and email
-  deleteMemberNameAndEmail(i : number): void{
-    this.memberNameEmail.removeAt(i);
+  deleteMemberNameAndEmailField(i: number): void {
```

Comment:
check and change method comments based on its format

Reply:
@param index......whats is the use or what does the index is used for.
For example: @params index Index of the user

Reply:
Why this change is not done yet?